### PR TITLE
Changes to fetch Region from Node Info

### PR DESCRIPTION
**Pull request by vasudha786**

### CODE REVIEW

**Comment by vasudha786** [comment]
1) Region in SC and Zone in SC

vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat sc1.yaml              
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
  labels:
    addonmanager.kubernetes.io/mode: Reconcile
    app: ibm-vpc-file-csi-driver
    razee/force-apply: "true"
  name: sc-region1
parameters:
  csi.storage.k8s.io/fstype: ext4
  encrypted: "false"
  encryptionKey: ""
  profile: "dp2"
  region: "us-east”
  resourceGroup: ""
  tags: ""
  zone: "us-east-1”
provisioner: vpc.file.csi.ibm.io
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer%                                                                                                                           vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f sc1.yaml
storageclass.storage.k8s.io/sc-region1 created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % vi pvc.yaml
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat pvc.yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: region-pvc-sc1
  #name: testpvc
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi
  storageClassName: sc-region1%                                                                                                                                    vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f pvc.yaml
persistentvolumeclaim/region-pvc-sc1 created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc | grep region-pvc-sc1
region-pvc-sc1   Pending                                      sc-region1     <unset>                 23s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc | grep region-pvc-sc1
region-pvc-sc1   Pending                                      sc-region1     <unset>                 42s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat pod.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: testpodrwm
  labels:
    app: testpodrwm
spec:
  replicas: 2
  selector:
    matchLabels:
      app: testpodrwm
  template:
    metadata:
      labels:
        app: testpodrwm
    spec:
      containers:
      - image: rabbitmq:3-management #image name which should be avilable within cluster
        name: rabbitmq # name of the container inside POD
        volumeMounts:
        - mountPath: /myvol  # mount path for pvc from container
          name: pvc-name # pvc name for this pod
      volumes:
      - name: pvc-name  # volume resource name in this POD, user can choose any name as per kubernetes
        persistentVolumeClaim:
          claimName: region-pvc-sc1 # pvc name which was created by using claim.yaml file
      nodeSelector:
        kubernetes.io/hostname: "10.241.0.4"
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f pod.yaml
deployment.apps/testpodrwm created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc | grep region-pvc-sc1
region-pvc-sc1   Pending                                      sc-region1     <unset>                 20s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc | grep region-pvc-sc1
region-pvc-sc1   Bound    pvc-93bc991a-8094-4cf1-a6eb-817f4bc95408   10Gi       RWO            sc-region1     <unset>                 77s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods
NAME                          READY   STATUS    RESTARTS   AGE
testpodrwm-849c56d9dc-pfrlb   1/1     Running   0          3m6s
testpodrwm-849c56d9dc-vg78t   1/1     Running   0          3m6s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get deployment
NAME         READY   UP-TO-DATE   AVAILABLE   AGE
testpodrwm   2/2     2            2           3m14s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec -it testpodrwm-849c56d9dc-pfrlb /bin/bash
error: exec [POD] [COMMAND] is not supported anymore. Use exec [POD] -- [COMMAND] instead
See 'kubectl exec -h' for help and examples
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec -it testpodrwm-849c56d9dc-pfrlb /bin/sh  
error: exec [POD] [COMMAND] is not supported anymore. Use exec [POD] -- [COMMAND] instead
See 'kubectl exec -h' for help and examples
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec testpodrwm-849c56d9dc-pfrlb -- /bin/sh 
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec -it testpodrwm-849c56d9dc-pfrlb -- /bin/sh
# cd myvol/                    
# echo "new" > hi.txt
# exit
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat myvol/hi.txt       
cat: myvol/hi.txt: No such file or directory
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec -it testpodrwm-849c56d9dc-pfrlb -- /bin/sh
# cat myvol/hi.txt
new

**Comment by vasudha786** [comment]
2) Region not in SC and Zone in SC

vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat sc2.yaml
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
  labels:
    addonmanager.kubernetes.io/mode: Reconcile
    app: ibm-vpc-file-csi-driver
    razee/force-apply: "true"
  name: sc-region2
parameters:
  csi.storage.k8s.io/fstype: ext4
  encrypted: "false"
  encryptionKey: ""
  profile: "dp2"
  region: ""
  resourceGroup: ""
  tags: ""
  zone: "us-east-1"
provisioner: vpc.file.csi.ibm.io
reclaimPolicy: Delete
volumeBindingMode: Immediate%                                                                                                                                                                 vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f sc2.yaml
storageclass.storage.k8s.io/sc-region2 created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat pvc.yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: region-pvc-sc2
  #name: testpvc
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi
  storageClassName: sc-region2%                                                                                                                                                               vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f pvc.yaml
persistentvolumeclaim/region-pvc-sc2 created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat pod.yaml              
apiVersion: apps/v1
kind: Deployment
metadata:
  name: testpodrwm1
  labels:
    app: testpodrwm1
spec:
  replicas: 2
  selector:
    matchLabels:
      app: testpodrwm1
  template:
    metadata:
      labels:
        app: testpodrwm1
    spec:
      containers:
      - image: rabbitmq:3-management #image name which should be avilable within cluster
        name: rabbitmq # name of the container inside POD
        volumeMounts:
        - mountPath: /myvol  # mount path for pvc from container
          name: pvc-name # pvc name for this pod
      volumes:
      - name: pvc-name  # volume resource name in this POD, user can choose any name as per kubernetes
        persistentVolumeClaim:
          claimName: region-pvc-sc2 # pvc name which was created by using claim.yaml file
      nodeSelector:
        kubernetes.io/hostname: "10.241.0.4"%                                                                                                                                                 vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f pod.yaml
deployment.apps/testpodrwm1 created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc | grep region-pvc-sc2
region-pvc-sc2   Bound    pvc-5d45a548-6d63-43b8-b0b4-98613d469b9c   10Gi       RWO            sc-region2     <unset>                 7m38s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl describe pv pvc-5d45a548-6d63-43b8-b0b4-98613d469b9c
Name:              pvc-5d45a548-6d63-43b8-b0b4-98613d469b9c
Labels:            <none>
Annotations:       pv.kubernetes.io/provisioned-by: vpc.file.csi.ibm.io
                   volume.kubernetes.io/provisioner-deletion-secret-name: 
                   volume.kubernetes.io/provisioner-deletion-secret-namespace: 
Finalizers:        [kubernetes.io/pv-protection]
StorageClass:      sc-region2
Status:            Bound
Claim:             default/region-pvc-sc2
Reclaim Policy:    Delete
Access Modes:      RWO
VolumeMode:        Filesystem
Capacity:          10Gi
Node Affinity:     
  Required Terms:  
    Term 0:        failure-domain.beta.kubernetes.io/region in [us-east]
Message:           
Source:
    Type:              CSI (a Container Storage Interface (CSI) volume source)
    Driver:            vpc.file.csi.ibm.io
    FSType:            ext4
    VolumeHandle:      r014-dedc0ac1-6718-4cea-8c73-0afb44c8af58#r014-4f27664e-c365-4d1f-8c8d-db376b07ea64
    ReadOnly:          false
    VolumeAttributes:      ENISecurityGroupIds=r014-3d3ea4b0-573d-4cd4-be86-af31827d9161
                           ENISubnetId=0757-aa155ba2-2870-47e0-bcc8-4526f80716cd
                           clusterID=csljm6tw0cvjfaim5m80
                           failure-domain.beta.kubernetes.io/region=us-east
                           fileShareId=r014-dedc0ac1-6718-4cea-8c73-0afb44c8af58
                           fileShareTargetId=r014-4f27664e-c365-4d1f-8c8d-db376b07ea64
                           iops=100
                           isENIEnabled=true
                           nfsServerPath=10.241.0.15:/c2878ca1_c11d_4e86_8d12_02605e2a3115
                           storage.kubernetes.io/csiProvisionerIdentity=1732086020680-8081-vpc.file.csi.ibm.io
                           tags=
                           volumeCRN=crn:v1:bluemix:public:is:us-east-1:a/47998ac029ed4ca69cf807338b7dbd2e::share:r014-dedc0ac1-6718-4cea-8c73-0afb44c8af58
                           volumeId=r014-dedc0ac1-6718-4cea-8c73-0afb44c8af58#r014-4f27664e-c365-4d1f-8c8d-db376b07ea64
Events:
  Type    Reason               Age    From                                          Message
  ----    ------               ----   ----                                          -------
  Normal  VolumeMetaDataSaved  7m25s  ibm-vpc-file-csi-controller-65b6d9d667-28nts  Success
  Normal  VolumeMetaDataSaved  7m22s  ibm-vpc-file-csi-controller-65b6d9d667-6p9s8  Success
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods 
NAME                           READY   STATUS    RESTARTS   AGE
testpodrwm-849c56d9dc-pfrlb    1/1     Running   0          156m
testpodrwm-849c56d9dc-vg78t    1/1     Running   0          156m
testpodrwm1-655c76db47-rbxgz   1/1     Running   0          2m1s
testpodrwm1-655c76db47-rh2m5   1/1     Running   0          2m1s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get deployments
NAME          READY   UP-TO-DATE   AVAILABLE   AGE
testpodrwm    2/2     2            2           157m
testpodrwm1   2/2     2            2           2m25s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec -it testpodrwm1-655c76db47-rbxgz -- /bin/bash
root@testpodrwm1-655c76db47-rbxgz:/# cd myvol/
root@testpodrwm1-655c76db47-rbxgz:/myvol# echo "new2" > hi.txt
root@testpodrwm1-655c76db47-rbxgz:/myvol# cat hi.txt
new2
root@testpodrwm1-655c76db47-rbxgz:/myvol# exit
exit

File: deployment.yaml
Comment (prankulmahajan) [inline deployment.yaml:1]: @vasudha786 Can you please remove this custom deployment file?

**Review by prankulmahajan — CHANGES_REQUESTED** [review]

File: deployment.yaml
Comment (vasudha786) [reply deployment.yaml:1]: I have deleted my deployment file

**Review by prankulmahajan — APPROVED** [review]
lgtm

File: pkg/ibmcsidriver/controller_test.go [hunk 1]
```diff
@@ -229,8 +229,16 @@ func TestCreateVolumeArguments(t *testing.T) {
 			expVol: &csi.Volume{
 				CapacityBytes:      20 * 1024 * 1024 * 1024, // In byte
 				VolumeId:           "testVolumeId" + VolumeIDSeperator + "testVolumeAccessPointId",
-				VolumeContext:      map[string]string{utils.NodeRegionLabel: "myregion", utils.NodeZoneLabel: "myzone", VolumeIDLabel: "testVolumeId" + VolumeIDSeperator + "testVolumeAccessPointId", FileShareIDLabel: "testVolumeId", FileShareTargetIDLabel: "testVolumeAccessPointId", IsENIEnabled: "false", NFSServerPath: "abc:/xyz/pqr", Tag: "", VolumeCRNLabel: "", ClusterIDLabel: "fake-cluster-id"},
+				VolumeContext:      map[string]string{utils.NodeRegionLabel: "testregion", utils.NodeZoneLabel: "myzone", VolumeIDLabel: "testVolumeId" + VolumeIDSeperator + "testVolumeAccessPointId", FileShareIDLabel: "testVolumeId", FileShareTargetIDLabel: "testVolumeAccessPointId", IsENIEnabled: "false", NFSServerPath: "abc:/xyz/pqr", Tag: "", VolumeCRNLabel: "", ClusterIDLabel: "fake-cluster-id"},
 				AccessibleTopology: stdTopology,
+				// CapacityBytes: 20 * 1024 * 1024 * 1024, // In byte
```
Comment (sameshai) [inline pkg/ibmcsidriver/controller_test.go:234]: Why the code is commented can we remove if not required

File: pkg/ibmcsidriver/controller_test.go [hunk 2]
```diff
@@ -278,8 +286,16 @@ func TestCreateVolumeArguments(t *testing.T) {
...
+				VolumeContext:      map[string]string{utils.NodeRegionLabel: "testregion", VolumeIDLabel: "testVolumeId" + VolumeIDSeperator + "testVolumeAccessPointId", FileShareIDLabel: "testVolumeId", FileShareTargetIDLabel: "testVolumeAccessPointId", IsENIEnabled: "true", ENISecurityGroupIDs: "kube-fake-cluster-id", ENISubnetID: "sub-1", NFSServerPath: "abc:/xyz/pqr", Tag: "", VolumeCRNLabel: "", ClusterIDLabel: "fake-cluster-id"},
 				AccessibleTopology: stdENITopology,
+				// CapacityBytes: 20 * 1024 * 1024 * 1024, // In byte
+				// VolumeId:      "testVolumeId" + VolumeIDSeperator + "testVolumeAccessPointId",
+				// VolumeContext: map[string]string{utils.NodeRegionLabel: "myregion", utils.NodeZoneLabel: "myzone", VolumeIDLabel: "testVolumeId" + VolumeIDSeperator + "testVolumeAccessPointId", FileShareIDLabel: "testVolumeId", FileShareTargetIDLabel: "testVolumeAccessPointId", IsENIEnabled: "true", ENISecurityGroupIDs: "kube-fake-cluster-id", ENISubnetID: "sub-1", NFSServerPath: "abc:/xyz/pqr", Tag: "", VolumeCRNLabel: "", ClusterIDLabel: "fake-cluster-id"},
```
Comment (sameshai) [inline pkg/ibmcsidriver/controller_test.go:293]: same Here

**Review by sameshai — CHANGES_REQUESTED** [review]
Given comments please check , 

can you check once why zone is not populate in this case , https://github.com/IBM/ibm-vpc-file-csi-driver/pull/90#issuecomment-2490162126 
 I see some gap

**Comment by prankulmahajan** [comment]
@vasudha786 Can you please address Sameer's comments and also do another test, Give region and not zone in sc

**Comment by vasudha786** [comment]
3. Region in SC and Zone not in SC
```
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat sc3.yaml                                               
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
  labels:
    addonmanager.kubernetes.io/mode: Reconcile
    app: ibm-vpc-file-csi-driver
    razee/force-apply: "true"
  name: sc-region3
parameters:
  csi.storage.k8s.io/fstype: ext4
  encrypted: "false"
  encryptionKey: ""
  profile: "dp2"
  region: "us-east"
  resourceGroup: ""
  tags: ""
  zone: ""
provisioner: vpc.file.csi.ibm.io
reclaimPolicy: Delete
volumeBindingMode: Immediate%                                                                                                      
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat pvc.yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: region-pvc-sc3
  #name: testpvc
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi
  storageClassName: sc-region3%                                                                                                    
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % cat pod.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: testpodrwm2
  labels:
    app: testpodrwm2
spec:
  replicas: 2
  selector:
    matchLabels:
      app: testpodrwm2
  template:
    metadata:
      labels:
        app: testpodrwm2
    spec:
      containers:
      - image: rabbitmq:3-management #image name which should be avilable within cluster
        name: rabbitmq # name of the container inside POD
        volumeMounts:
        - mountPath: /myvol  # mount path for pvc from container
          name: pvc-name # pvc name for this pod
      volumes:
      - name: pvc-name  # volume resource name in this POD, user can choose any name as per kubernetes
        persistentVolumeClaim:
          claimName: region-pvc-sc3 # pvc name which was created by using claim.yaml file
      nodeSelector:
        kubernetes.io/hostname: "10.241.0.4"%                                                                                      
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f sc3.yaml         
storageclass.storage.k8s.io/sc-region3 created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f pvc.yaml
persistentvolumeclaim/region-pvc-sc3 created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc | grep region-pvc-sc3
region-pvc-sc3   Pending                                                                        sc-region3     <unset>                 5s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc | grep region-pvc-sc3
region-pvc-sc3   Pending                                                                        sc-region3     <unset>                 36s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pvc | grep region-pvc-sc3
region-pvc-sc3   Bound    pvc-27756be2-8630-4f9d-9d7c-8fd02231f515   10Gi       RWO            sc-region3     <unset>                 51s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl describe pvc region-pvc-sc3
Name:          region-pvc-sc3
Namespace:     default
StorageClass:  sc-region3
Status:        Bound
Volume:        pvc-9c798399-1775-47b1-b737-12acfb082b13
Labels:        <none>
Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: vpc.file.csi.ibm.io
               volume.kubernetes.io/storage-provisioner: vpc.file.csi.ibm.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      10Gi
Access Modes:  RWO
VolumeMode:    Filesystem
Used By:       testpodrwm2-f6b5bbd5c-5dgz6
               testpodrwm2-f6b5bbd5c-672bc
Events:
  Type    Reason                 Age                 From                                                                                                   Message
  ----    ------                 ----                ----                                                                                                   -------
  Normal  Provisioning           105s                vpc.file.csi.ibm.io_ibm-vpc-file-csi-controller-65b6d9d667-28nts_0c69f40e-8313-4f29-899e-395686c8916e  External provisioner is provisioning volume for claim "default/region-pvc-sc3"
  Normal  ExternalProvisioning   81s (x4 over 105s)  persistentvolume-controller                                                                            Waiting for a volume to be created either by the external provisioner 'vpc.file.csi.ibm.io' or manually by the system administrator. If volume creation is delayed, please verify that the provisioner is running and correctly registered.
  Normal  ProvisioningSucceeded  76s                 vpc.file.csi.ibm.io_ibm-vpc-file-csi-controller-65b6d9d667-28nts_0c69f40e-8313-4f29-899e-395686c8916e  Successfully provisioned volume pvc-9c798399-1775-47b1-b737-12acfb082b13
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl describe pv pvc-9c798399-1775-47b1-b737-12acfb082b13
Name:              pvc-9c798399-1775-47b1-b737-12acfb082b13
Labels:            <none>
Annotations:       pv.kubernetes.io/provisioned-by: vpc.file.csi.ibm.io
                   volume.kubernetes.io/provisioner-deletion-secret-name: 
                   volume.kubernetes.io/provisioner-deletion-secret-namespace: 
Finalizers:        [kubernetes.io/pv-protection]
StorageClass:      sc-region3
Status:            Bound
Claim:             default/region-pvc-sc3
Reclaim Policy:    Delete
Access Modes:      RWO
VolumeMode:        Filesystem
Capacity:          10Gi
Node Affinity:     
  Required Terms:  
    Term 0:        failure-domain.beta.kubernetes.io/region in [us-east]
Message:           
Source:
    Type:              CSI (a Container Storage Interface (CSI) volume source)
    Driver:            vpc.file.csi.ibm.io
    FSType:            ext4
    VolumeHandle:      r014-e70e0594-fa80-475a-8b21-d37e7ae8e8e3#r014-ca0d172f-d4d7-4807-a00a-f6400815baa3
    ReadOnly:          false
    VolumeAttributes:      ENISecurityGroupIds=r014-3d3ea4b0-573d-4cd4-be86-af31827d9161
                           ENISubnetId=0777-86b7ef18-6373-437f-a935-1a248b0c3813
                           clusterID=csljm6tw0cvjfaim5m80
                           failure-domain.beta.kubernetes.io/region=us-east
                           fileShareId=r014-e70e0594-fa80-475a-8b21-d37e7ae8e8e3
                           fileShareTargetId=r014-ca0d172f-d4d7-4807-a00a-f6400815baa3
                           iops=100
                           isENIEnabled=true
                           nfsServerPath=10.241.128.18:/86c8000f_a2a2_418f_8092_b79b4061e5e1
                           storage.kubernetes.io/csiProvisionerIdentity=1732086020680-8081-vpc.file.csi.ibm.io
                           tags=
                           volumeCRN=crn:v1:bluemix:public:is:us-east-3:a/47998ac029ed4ca69cf807338b7dbd2e::share:r014-e70e0594-fa80-475a-8b21-d37e7ae8e8e3
                           volumeId=r014-e70e0594-fa80-475a-8b21-d37e7ae8e8e3#r014-ca0d172f-d4d7-4807-a00a-f6400815baa3
Events:
  Type    Reason               Age   From                                          Message
  ----    ------               ----  ----                                          -------
  Normal  VolumeMetaDataSaved  109s  ibm-vpc-file-csi-controller-65b6d9d667-6p9s8  Success
  Normal  VolumeMetaDataSaved  106s  ibm-vpc-file-csi-controller-65b6d9d667-28nts  Success
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl create -f pod.yaml
deployment.apps/testpodrwm2 created
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get pods
NAME                           READY   STATUS    RESTARTS   AGE
testpodrwm-849c56d9dc-pfrlb    1/1     Running   0          11d
testpodrwm-849c56d9dc-vg78t    1/1     Running   0          11d
testpodrwm1-655c76db47-rbxgz   1/1     Running   0          11d
testpodrwm1-655c76db47-rh2m5   1/1     Running   0          11d
testpodrwm2-f6b5bbd5c-bcmlz    1/1     Running   0          24s
testpodrwm2-f6b5bbd5c-tzgll    1/1     Running   0          24s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl get deployments
NAME          READY   UP-TO-DATE   AVAILABLE   AGE
testpodrwm    2/2     2            2           11d
testpodrwm1   2/2     2            2           11d
testpodrwm2   2/2     2            2           33s
vasudha@Vasudhas-MacBook-Pro ibm-vpc-file-csi-driver % kubectl exec -it testpodrwm2-f6b5bbd5c-tzgll -- /bin/sh
# cd myvol/
# echo "new3" > hi.txt
# cat hi.txt
new3
# exit
```

**Review by prankulmahajan — APPROVED** [review]
lgtm. Test results looks good.
Will test it in stage as well

**Review by sameshai — APPROVED** [review]
lgtm

**Comment by sean-freeman** [comment]
@vasudha786 This alters the deployment code with below block, please also alter in secondary deployment path for OCP Self-Managed to keep both aligned as code changes are made to the underlying driver.....

[deploy/openshift-self-managed/driver/kubernetes/manifests/controller-server.yaml#L127](https://github.com/IBM/ibm-vpc-file-csi-driver/blob/master/deploy/openshift-self-managed/driver/kubernetes/manifests/controller-server.yaml#L127)


```
            - name: KUBE_NODE_NAME
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
```

**Comment by prankulmahajan** [comment]
> @vasudha786 This alters the deployment code with below block, please also alter in secondary deployment path for OCP Self-Managed to keep both aligned as code changes are made to the underlying driver.....
> 
> [deploy/openshift-self-managed/driver/kubernetes/manifests/controller-server.yaml#L127](https://github.com/IBM/ibm-vpc-file-csi-driver/blob/master/deploy/openshift-self-managed/driver/kubernetes/manifests/controller-server.yaml#L127)
> 
> ```
>             - name: KUBE_NODE_NAME
>               valueFrom:
>                 fieldRef:
>                   fieldPath: spec.nodeName
> ```

Sure @sean-freeman , let me open a PR for the same. We will manage both path from now

**Pull request merged**